packages/zero-cache/src/server/inspector-delegate.ts

```diff
@@ -50,24 +53,35 @@ export class InspectMetricsDelegate implements MetricsDelegate {
     return mapValues(this.#globalMetrics, v => v.toJSON());
   }
 
-  deleteMetricsForQuery(queryID: string): void {
+  getASTForQuery(queryID: string): AST | undefined {
+    const transformationHash = this.#queryIDToTransformationHash.get(queryID);
+    return transformationHash
+      ? this.#transformationASTs.get(transformationHash)
+      : undefined;
+  }
+
+  removeQuery(queryID: string): void {
     this.#perQueryServerMetrics.delete(queryID);
+    this.#queryIDToTransformationHash.delete(queryID);
     // Remove queryID from all hash-to-ID mappings
-    for (const [hash, idSet] of this.#hashToIDs.entries()) {
+    for (const [transformationHash, idSet] of this.#hashToIDs.entries()) {
```

Comment:
do we need to iterate the whole map? Wouldn't we have the specific transformation hash here from the line above:

```
this.#queryIDToTransformationHash.delete(queryID);
```